app/src/Containers/EventsContainer.tsx

```diff
@@ -123,23 +133,37 @@ const EventsContainer = (props: {
     )
   }
 
-  if (allUpcomingEvents || eventsSearch) {
-    const eventsToShow = eventsSearch
-      ? eventsSearch.results
-      : (allUpcomingEvents as Events)
+  const [eventsToShow, setEventsToShow] = useState<Events>() // holding state for which event are being rendered
 
-    return (
-      <Theme>
-        <EventList data={eventsToShow} />
-      </Theme>
-    )
-  } else {
-    return (
+  useEffect (() => {
+    eventToShow()
+  }, [allUpcomingEvents, [props.route.params]])
+
+  const eventToShow = () => {
+    if(eventsSearch){ 
+      setEventsToShow(eventsSearch.results)
+    } else if (!eventsSearch && allUpcomingEvents && selectedOption === EventsRange.Upcoming){
+      setEventsToShow(allUpcomingEvents as Events)
+    } else if (!eventsSearch && allUpcomingEvents && allPastEvents && selectedOption === EventsRange.Past){
+      setEventsToShow(allPastEvents as Events)
+    }
+  } // Logic for deciding which event to render
+
+  if(eventsToShow){
+    if (allUpcomingEvents || eventsSearch) {
```

Comment:
I don't think the logic on this line (and its closing bracket on line 159) are needed.  (Did you plan to include an `else` for past events which you've removed?)